packages/teams-js/src/public/validatedStringId.ts

```diff
@@ -0,0 +1,34 @@
+import { validateSafeContent } from '../internal/idValidation';
+import { ISerializable } from './serializable.interface';
+
+/**
+ * A strongly typed class used to represent a "valid" string id.
+ */
+export class ValidatedStringId implements ISerializable {
+  /**
+   * Creates a strongly-typed Id from a string
+   *
+   * @param idAsString An id represented as a string
+   * @throws Error with a message describing the exact validation violation
+   */
+  public constructor(private readonly idAsString: string) {
+    validateSafeContent(idAsString);
```

Comment:
Why doesn't `ValidatedStringId` validate the string length too?

Reply:
During my output-validation conversation with security and Teams, it came up that min/max is an old recommendation that stems from C/C++ code where developers had to deal with fixed memory space, reserved in advance, to store property values. And thus, they had to make sure that an incoming property value would not exceed the length they reserved memory for, otherwise that could cause a buffer overrun which used to be a massive attack vector in Windows prior to XP Service Pack 2.
Security was ok to not have the min max length requirement. For that reason, I created a new class that just checks for Opaque and script tags. 

Your comment made me thought that we can have a uuid instead. I will check with 3pcxp and will respond back.

Reply:
@ndangudubiyyam Niharika, I am going to update 3pcxp. If you agree with this decision to use UUID, can you update hubsdk PR please? I will report here what partners response, but you can keep track of the convo in Teams channel.

Reply:
Please make sure it's really a UUID! We always get tripped up when it's something that _looks_ like a UUID but doesn't actually confirm to the UUID spec

Reply:
We already have a class UUID that validate that the id is uuid type. I am planning to use that

Reply:
@jekloudaMSFT if we do go with UUID, do you want me to delete the validatedStringId class? OR is it still beneficial to have it around for future use?

Reply:
I think it's still beneficial to have around, if you wanted to make it a little more generic, it could be `validatedSafeString` or something, since there's nothing really ID specific in the implementation, right? But I think it's also fine to leave it as is

Reply:
You are right there is nothing special that makes the string ID. I will take the name change suggestion!